packages/vscode-extension-serverless-workflow-editor/src/extension/builtInVsCodeEditorSwfContributions.ts

```diff
@@ -66,6 +67,27 @@ export function setupBuiltInVsCodeEditorSwfContributions(args: {
     },
   };
 
+  const swfJsonValidationDiganosticsCollection: vscode.DiagnosticCollection =
+    vscode.languages.createDiagnosticCollection("SWF-JSON-VALIDATION-DIAGNOSTICS-COLLECTION");
+
+  args.context.subscriptions.push(
+    vscode.workspace.onDidOpenTextDocument(async (doc: vscode.TextDocument) => {
+      setSwfJsonDiagnostics(args, doc.uri, swfJsonValidationDiganosticsCollection);
+    })
+  );
+
+  args.context.subscriptions.push(
+    vscode.workspace.onDidChangeTextDocument(async (event: vscode.TextDocumentChangeEvent) => {
+      const doValidationOnChange = debounce(
+        (args: ls.Command, uri: vscode.Uri, swfJsonValidationDiganosticsCollection: vscode.DiagnosticCollection) => {
+          setSwfJsonDiagnostics(args, uri, swfJsonValidationDiganosticsCollection);
+        },
+        1000
+      );
```

Comment:
Maybe we can do this shorter? Like 
```suggestion
      const doValidationOnChange = debounce(setSwfJsonDiagnostics, 1000);
```